src/C-interface/ellpack/bml_multiply_ellpack_typed.c

```diff
@@ -289,100 +280,91 @@ void TYPED_FUNC(
     int rowMin = A_localRowMin[myRank];
     int rowMax = A_localRowMax[myRank];
 
-#if !(defined(__IBMC__) || defined(__ibmxl__))
-    int ix[C->N], jx[C->N];
-    REAL_T x[C->N];
+    int nteams = MIN(BML_TEAMS, rowMax - rowMin + 1);
 
-    memset(ix, 0, C->N * sizeof(int));
-    memset(jx, 0, C->N * sizeof(int));
-    memset(x, 0.0, C->N * sizeof(REAL_T));
-#endif
+    int all_ix[C->N * nteams], all_jx[C->N * nteams];
+    REAL_T all_x[C->N * nteams];
+
+    memset(all_ix, 0, C->N * nteams * sizeof(int));
+    memset(all_jx, 0, C->N * nteams * sizeof(int));
+    memset(all_x, 0.0, C->N * nteams * sizeof(REAL_T));
 
 #if defined (USE_OMP_OFFLOAD)
-#pragma omp target teams distribute parallel for \
-    shared(A_N, A_M, A_nnz, A_index, A_value)  \
-    shared(A_localRowMin, A_localRowMax)       \
-    shared(B_N, B_M, B_nnz, B_index, B_value)  \
-    shared(C_N, C_M, C_nnz, C_index, C_value)  \
-    firstprivate(ix, jx, x)
-#else
-#if defined(__IBMC__) || defined(__ibmxl__)
+#pragma omp target map(to:all_ix[0:C_N*nteams],all_jx[0:C_N*nteams],all_x[0:C_N*nteams])
```

Comment:
and here

Reply:
Apologies, this isn't ready for review. I'm just capturing the state before I incorporate the changes discussed in today's meeting about the Iris build, which will be fairly extensive.

Let me know if I should close the pull request or keep it open while I revise.

Reply:
You could convert the PR to a draft while working on it. I would prefer that over closing it altogether.

Reply:
@nicolasbock @jeanlucf22 Done, thanks.

Reply:
@nicolasbock @jeanlucf22 This is now ready for review.